### PR TITLE
Release/0.2.1

### PR DESCRIPTION
**Pull request by ThinkTankShark**

### CODE REVIEW

File: lib/wipe-dependencies.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const fs = require('fs');
```
Comment (isuttell) [inline lib/wipe-dependencies.js]: What is this file? There's no comments or explanation

File: package.json [hunk 2]
```diff
@@ -1,84 +1,107 @@
...
+    "es2015",
+    "javascript",
+    "javascript-library",
+    "grunt",
+    "webpack",
+    "jest"
   ],
-  "author": "Isaac Suttell <isaac@isaacsuttell.com>",
+  "author": "Chris Orescan <chris.orescan@sony.com>",
+  "contributor": "Sepand Assadi <sepand.assadi@sony.com>",
```
Comment (isuttell) [inline package.json]: I think it this needs to be an array and plural

File: package.json [hunk 3]
```diff
@@ -1,84 +1,107 @@
...
+    "file-loader": "^0.11.2",
+    "grunt": "^1.0.1",
+    "grunt-contrib-clean": "^1.1.0",
+    "grunt-coveralls": "^1.0.1",
+    "grunt-eslint": "^20.0.0",
+    "grunt-webpack": "^3.0.0",
+    "jest": "^20.0.4",
+    "jest-css-modules": "^1.1.0",
+    "lodash": "^4.17.4",
+    "object-assign": "^4.1.1",
```
Comment (isuttell) [inline package.json]: Do we really need object-assign? and lodash?

File: src/Controls.js [hunk 4]
```diff
@@ -1,62 +1,103 @@
...
 export default class SelectControls extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleOpenDropdown = this.handleOpenDropdown.bind(this);
+  }
 
+  /**
+   * Enables the focus on input DOM
+   *
+   * @memberof SelectControls
```
Comment (isuttell) [inline src/Controls.js]: What does this `@memberof` do?

File: src/Tag.js [hunk 5]
```diff
@@ -1,18 +1,35 @@
...
 
+  /**
+   * Optimizes the app performance
+   *
+   * @param {object} nextProps
+   * @returns {bool}
+   * @memberof Tag
+   */
+  shouldComponentUpdate(nextProps) {
+    return this.props !== nextProps;
```
Comment (isuttell) [inline src/Tag.js]: props is an native object, you can't directly compare

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import TagContainer from './TagContainer';
+
+export default class TagInput extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      selection: []
```
Comment (isuttell) [inline src/index.js]: Can we use immutables instead?

File: README.md [hunk 7]
```diff
@@ -1,61 +1,170 @@
 # ship-components-tag-input
-[React](http://facebook.github.io/react/) select box. Exports a commonjs module that can be used with [webpack](http://webpack.github.io/). Source is in ES6 and is compiled down to ES5 using [Babel](https://babeljs.io/).
 
-[![npm](https://img.shields.io/npm/v/ship-components-select.svg?maxAge=2592000)](https://www.npmjs.com/package/ship-components-select)
+[JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript) Material Design React Multi-Select Box. Exports a commonjs module that can be used with [webpack](http://webpack.github.io/). Source is in ES6 and an ES5 version is available using [Babel](https://babeljs.io/).
+
+[![npm](https://img.shields.io/npm/v/ship-components-utility.svg?maxAge=2592000)](https://www.npmjs.com/package/ship-components-tag-input)
```
Comment (ThinkTankShark) [inline README.md]: Nice catch.

File: README.md [hunk 8]
```diff
@@ -1,61 +1,170 @@
...
 ```
-
 which will live reload any changes you make and serve them at http://localhost:8080.
 
 ### Webpack Configuration
 This module is designed to be used with webpack but requires a few loaders if you are pulling the source into another project.
 
 ```shell
-$ npm install webpack babel-loader css-loader style-loader postcss-loader extract-text-webpack-plugin postcss-nested postcss-color-hex-alpha postcss-color-function postcss-calc autoprefixer --save-dev
+$ npm install autoprefixer babel-core babel-eslint babel-loader babel-plugin-transform-runtime babel-preset-env babel-preset-react coveralls css-loader eslint eslint-config-ship-components eslint-loader eslint-plugin-react extract-text-webpack-plugin file-loader grunt grunt-contrib-clean grunt-coveralls grunt-eslint grunt-webpack jest jest-css-modules lodash object-assign package-banner postcss-calc postcss-color-function postcss-color-hex-alpha postcss-loader postcss-nested postcss-simple-vars prop-types react-addons-test-utils sockjs-client style-loader webpack webpack-dev-server --save-dev
```
Comment (ThinkTankShark) [inline README.md]: none of them, removed.

File: lib/wipe-dependencies.js (hunk 1 above)
Comment (ThinkTankShark) [inline lib/wipe-dependencies.js]: I added the comments.... It's basically a helper function to update all dependencies as an optional task...

File: package.json (hunk 2 above)
Comment (ThinkTankShark) [inline package.json]: Fixed

File: package.json (hunk 3 above)
Comment (ThinkTankShark) [inline package.json]: We don't, I removed both from the code but forgot to uninstall  / remove from package

File: src/Controls.js (hunk 4 above)
Comment (ThinkTankShark) [inline src/Controls.js]: Documentation purposes, more info here: http://usejsdoc.org/tags-memberof.html

File: src/Tag.js (hunk 5 above)
Comment (ThinkTankShark) [inline src/Tag.js]: Right, fixed.

File: src/index.js (hunk 6 above)
Comment (ThinkTankShark) [inline src/index.js]: Done

File: lib/wipe-dependencies.js (hunk 1 above)
Comment (isuttell) [inline lib/wipe-dependencies.js]: I think that's better to leave outside the repo. You often don't want to update everything too. I typically use [npm-check](https://www.npmjs.com/package/npm-check) which lets you pick and choose what you want to update.

File: package.json [hunk 9]
```diff
@@ -1,84 +1,110 @@
...
+    "watch:test": "jest --coverage --watch",
+    "coverage": "cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js -v",
+    "build": "npm test && grunt clean && grunt build"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/ship-components/ship-components-tag-input.git"
   },
+  "engines": {
+    "node": ">=4.0 <6.0"
```
Comment (isuttell) [inline package.json]: We support 6, this excludes 6 =)

**Pull request merged**